test/utils/get_localize_status_test.dart

```diff
@@ -0,0 +1,542 @@
+
+      expect(textWidget.data, equals('online 10m ago'));
+    });
+
+    testWidgets(
+        'GIVEN the presence time to be 10 minute from present with currently active be false\n'
+        'THEN should display the status as online 10 minutes ago\n',
+        (WidgetTester tester) async {
+      // Given
+      final lessThan10MinutesAgo = (const Duration(minutes: 10)).inMilliseconds;
+      final presence = CachedPresence(
+        PresenceType.offline,
+        lessThan10MinutesAgo,
+        "",
+        false,
+        "testuserid",
+      );
+      final client = MockClient();
+      final room = Room(id: 'testid', client: client);
+
+      // WHEN
+      final textWidgetBuilder = Builder(
+        builder: (BuildContext context) {
+          final displayText =
+              room.getLocalizedStatusDirectChat(presence, context);
+          return Text(
+            key: textWidgetKey,
+            displayText,
+          );
+        },
+      );
+
+      await tester.pumpWidget(
+        MaterialApp(
+          locale: const Locale('en'),
+          localizationsDelegates: L10n.localizationsDelegates,
+          supportedLocales: L10n.supportedLocales,
+          home: textWidgetBuilder,
+        ),
+      );
+
+      final textWidgetFinder = find.byKey(textWidgetKey);
+
+      expect(textWidgetFinder, findsOneWidget);
+
+      final Text textWidget = tester.widget(textWidgetFinder) as Text;
+
+      expect(textWidget.data, isNotNull);
+
+      expect(textWidget.data, equals('online 10m ago'));
+    });
+
+    testWidgets(
+        'GIVEN the presence time to be 10 minute from present with currently active be false\n'
+        'THEN should display the status as online 10 minutes ago\n',
+        (WidgetTester tester) async {
+      // Given
+      final lessThan20hoursAgo = (const Duration(hours: 20)).inMilliseconds;
+      final presence = CachedPresence(
+        PresenceType.offline,
+        lessThan20hoursAgo,
+        "",
+        false,
+        "testuserid",
+      );
+      final client = MockClient();
+      final room = Room(id: 'testid', client: client);
+
+      // WHEN
+      final textWidgetBuilder = Builder(
+        builder: (BuildContext context) {
+          final displayText =
+              room.getLocalizedStatusDirectChat(presence, context);
+          return Text(
+            key: textWidgetKey,
+            displayText,
+          );
+        },
+      );
+
+      await tester.pumpWidget(
+        MaterialApp(
+          locale: const Locale('en'),
+          localizationsDelegates: L10n.localizationsDelegates,
+          supportedLocales: L10n.supportedLocales,
+          home: textWidgetBuilder,
+        ),
+      );
+
+      final textWidgetFinder = find.byKey(textWidgetKey);
+
+      expect(textWidgetFinder, findsOneWidget);
+
+      final Text textWidget = tester.widget(textWidgetFinder) as Text;
+
+      expect(textWidget.data, isNotNull);
+
+      expect(textWidget.data, equals('online 20h ago'));
```

Comment:
All tests have similar steps, I think you can create a common function. WDYT?

Reply:
okay i create a function for WHEN part, other part still the same to keep the format of the test for easier to follow